syntax = "proto3";

package greet;


message Parameters {
    repeated bytes tensors = 1;
    string tensor_type = 2;
}

message HelloRequest {
    Parameters parameters = 1;
}

message HelloReply {
    Parameters parameters = 1;
}

message DelayedReply {
    string message = 1;
    repeated HelloRequest request = 3;
}

message TestRequest {
    string msg = 1;
}

message TestReply {
    string msg = 1;
}

service Greeter {
    // Unary
    rpc SayHello (stream TestRequest) returns (stream TestReply);

    // Server Streaming
    rpc ParrotSaysHello (HelloRequest) returns (stream HelloReply);

    // CLient Streaming
    rpc ChattyClientSaysHello (stream HelloRequest) returns (DelayedReply);

    // Both Streaming
    rpc InteractingHello (stream HelloRequest) returns (stream HelloReply);
}